### ISSUE TITLE
Allow running multiples scripts in the same time

### ISSUE THREAD

**Issue by csphoenix1** [bugy/script-server#84]
It could be usefull to allow us running multiples scripts as the same time.
Usecase : running a script during another one is finishing to run.

Thank you :) It's realy a good project !!

**Comment by bugy** [comment]
Hi @csphoenix1 thank you for the issue and positive feedback. 

Couple of questions to better understand the usecase:
1. How many scripts do you usually run in parallel?
2. Are these the same scripts?
3. Could script executions influence each other?
4. You can "workaround" the problem by opening multiple tabs of script server, right?
5. Do you have an suggestion, how you would like to see running process? (which information you need, where to locate, etc.)

I'll try to implement the feature in December.

**Comment by csphoenix1** [comment]
Hi @bugy ,

1. For this moment, it's just few scripts, but, if that's possible, I could plan to execute maybe 20 scripts in the same time ? :) (cause I have some scripts which make a long time to be completed).
1. Effectively, they could have multiple instance of the same scripts.
1. I have no script which could influence each other, but, if it append, is that a problem?
1. Yes, it's what I do, but when I create this issue, I was thinking about a new feature that allow us to control the execution process of the instances that running. Something like "portainer" for Docker :)
1. I absolutely don't know. Maybe... when you click on a script in the left panel, we could have then multiple tabs on the left panel to execute a new (or multiples new) instances of the script in order to see the logs of the differents instances and allow us to interact with then. I don't know if that is easily possible.

I hope that answer to your questions.
PS : I'm sorry about my bad english ^^
PS² : thank you a lot again for this projet, It's really really a good tool. I've created a Dockerfile if it can help you, I will post it soon if you want :)

Good bye.

Regards,

**Comment by bugy** [comment]
Hi @csphoenix1, thanks for the answers. I'll see how would it better to implement it.
And I'll have a look at "portainer" and other similar tools

**Comment by bugy** [comment]
Hi @csphoenix1, I've implemented the first (very simple) version. Now a user will be able to run _different_ scripts in parallel. Simultaneous execution of the same script is not possible yet.
Could you check if current change works for you? Do you need parallel execution of same scripts? Unfortunately I couldn't come up with a clean design for such scenario

*Important* you have to _init.py_ again in order to download new library (font)

PS could you email me the dockerfile please? I think it would be useful for other people as well, because not everyone has python3

**Comment by csphoenix1** [comment]
Hi @bugy !
First, best wishes for this new year :)
About your implementation, it work perfectly!
I've test with some different scripts.
It's really cool (the loading round, the check after the script is completed, ...).
Thank you!
(I will analyse your code to learn how you did it :) )

Ps : no problem. I will send you the Dockerfile but just before: I have to test it to be sure it work really well.  Then I'll send it to you :)

**Comment by bugy** [comment]
Hi @csphoenix1, happy new year to you too :)
Thank you for the feedback and contribution!

I'm closing the ticket as finished then. If you have any further questions/ideas, please don't hesitate to contact me or to raise new issue.
